examples/Example/Page/Todo.hs

```diff
@@ -133,34 +138,31 @@ instance (Todos :> es) => HyperView TodoView es where
   type Require TodoView = '[AllTodos]
 
   data Action TodoView
-    = SetCompleted Todo Bool
-    | Edit Todo
-    | SubmitEdit Todo
+    = Edit FilterTodo Todo
+    | SubmitEdit FilterTodo Todo
     deriving (Show, Read, ViewAction)
 
-  update (SetCompleted todo completed) = do
-    updated <- Todos.setCompleted completed todo
-    pure $ todoView updated
-  update (Edit todo) = do
-    pure $ todoEditView todo
-  update (SubmitEdit todo) = do
+  update (Edit filt todo) = do
+    pure $ todoEditView filt todo
+  update (SubmitEdit filt todo) = do
     TodoForm task <- formData @(TodoForm Identity)
     updated <- Todos.setTask task todo
-    pure $ todoView updated
+    pure $ todoView filt updated
 
-todoView :: Todo -> View TodoView ()
-todoView todo = do
+todoView :: FilterTodo -> Todo -> View TodoView ()
+todoView filt todo = do
   row (border (TRBL 0 0 1 0) . pad 10) $ do
-    toggleCheckBtn (SetCompleted todo) todo.completed
-    el (completed . pad (XY 18 4) . onDblClick (Edit todo)) $ text todo.task
+    target AllTodos $ do
+      toggleCheckBtn (SetCompleted filt todo) todo.completed
+    el (completed . pad (XY 18 4) . onDblClick (Edit filt todo)) $ text todo.task
  where
   completed = if todo.completed then Style.strikethrough else id
 
-todoEditView :: Todo -> View TodoView ()
-todoEditView todo = do
+todoEditView :: FilterTodo -> Todo -> View TodoView ()
```

Comment:
Just food for thought, but since you mentioned that passing around the filter is annoying, I'm wondering if it would be possible to express a "model" (elm-style) that would typically go with a hyperview instance and be automatically serialized/deserialized in the update function (backed by a cookie store I suppose?)

Maybe I missed it, but it doesn't look like we can rely on data to be present in the browser's. So maybe that approach would make the "trigger" proposition not needed (which I don't fully grasp yet)

Reply:
Avoiding a model was a difficult design decision. Ultimately it wasn’t worth the cost. It’s less annoying to pass small bits of state around than to build in a model. 

The design philosophy of Hyperbole is closer to HTMX than it is to Elm. I think it’s a good trade off, both making the framework more intuitive, and I think it’s easier to work with for experienced users too

Reply:
> Maybe I missed it, but it doesn't look like we can rely on data to be present in the browser's.

I don’t understand, what do you mean by data being present in the browser? Hyperbole has built in support for state persistence via cookies/sessions, query parameters, and passing state in ViewActions. Is that what you’re talking about?

Reply:
Oh I see you’re talking about persisting a Model. An old version of this years ago kept track by serializing it, then passing it back to the server on a request. It worked great, but I decided to go with a simpler Action -> Updated View for this. Instead of Action -> Reduce Model -> Updated View. 

It’s simpler, easier to work with so far (I worked in Elm, React, and others for years)

Reply:
Yes, I was talking about persisting the Model. Sounds good to me though, I'm more familiar with Elm than anything, so I have to readjust my mental model a bit :+1: